Model/ConfigurationTracker.php

```diff
@@ -0,0 +1,128 @@
+use Algolia\AlgoliaSearch\Setup\UpgradeSchema;
+
+
+class ConfigurationTracker
+{
+    /** @var Data */
+    private $proxyHelper;
+
+    /** @var ConfigHelper */
+    private $configHelper;
+
+    /** @var QueryCollection */
+    private $queryCollection;
+
+    /** @var LandingPageCollection */
+    private $landingPageCollection;
+
+    /** @var UpgradeSchema */
+    private $upgradeSchema;
+
+    /**
+     * @param ProxyHelper $proxyHelper
+     * @param ConfigHelper $configHelper
+     * @param QueryCollection $queryCollection
+     * @param LandingPageCollection $landingPageCollection
+     */
+    public function __construct(
+        ProxyHelper $proxyHelper,
+        ConfigHelper $configHelper,
+        QueryCollection $queryCollection,
+        LandingPageCollection $landingPageCollection,
+        UpgradeSchema $upgradeSchema
+    ) {
+        $this->proxyHelper = $proxyHelper;
+        $this->configHelper = $configHelper;
+        $this->queryCollection = $queryCollection;
+        $this->landingPageCollection = $landingPageCollection;
+        $this->upgradeSchema = $upgradeSchema;
+    }
+
+    /**
+     * @param int $storeId
+     */
+    public function trackConfiguration($storeId)
+    {
+        $this->proxyHelper->trackEvent($this->configHelper->getApplicationID($storeId), 'Configuration saved', [
+            'source' => 'magento2.saveconfig',
+            'indexingEnabled' => $this->configHelper->isEnabledBackend($storeId),
+            'searchEnabled' => $this->configHelper->isEnabledFrontEnd($storeId),
+            'autocompleteEnabled' => $this->configHelper->isAutoCompleteEnabled($storeId),
+            'instantsearchEnabled' => $this->configHelper->isInstantEnabled($storeId),
+            'sortingChanged' => $this->isSortingChanged($storeId),
+            'rankingChanged' => $this->isCustomRankingChanged($storeId),
+            'replaceImageByVariantUsed' => $this->configHelper->useAdaptiveImage($storeId),
+            'indexingQueueEnabled' => $this->configHelper->isQueueActive($storeId),
+            'synonymsManagementEnabled' => $this->configHelper->isEnabledSynonyms($storeId),
+            'clickAnalyticsEnabled' => $this->configHelper->isClickConversionAnalyticsEnabled($storeId),
+            'googleAnalyticsEnabled' => $this->configHelper->isAnalyticsEnabled($storeId),
+            'customerGroupsEnabled' => $this->configHelper->isCustomerGroupsEnabled($storeId),
+            'merchangisingQRsCreated' => $this->getCountMerchandisingQueries() > 0,
+            'noOfMerchandisingQRs' => (int) $this->getCountMerchandisingQueries(),
+            'landingPageCreated' => $this->getCountLandingPages() > 0,
+            'noOfLandingPages' => (int) $this->getCountLandingPages(),
+            'storeId' => $storeId
+        ]);
+    }
+
+    /**
+     * @param int $storeId
+     *
+     * @return bool
+     */
+    private function isSortingChanged($storeId)
```

Comment:
The parent method of this function has the same parameter that is required. 

' public function trackConfiguration($storeId)`

I can add the option for default but it will always have a $storeId param in this case.